src/main/java/nl/dtls/fairdatapoint/service/impl/PurlPIDSystemImpl.java

```diff
@@ -0,0 +1,106 @@
+import nl.dtl.fairmetadata4j.model.FDPMetadata;
+import nl.dtl.fairmetadata4j.model.Metadata;
+import nl.dtls.fairdatapoint.service.PIDSystem;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+/**
+ * Implementation of purl.org PID system
+ *
+ * @author Rajaram Kaliyaperumal <rr.kaliyaperumal@gmail.com>
+ * @author Kees Burger <kees.burger@dtls.nl>
+ * @since 2018-06-06
+ * @version 0.1
+ */
+public class PurlPIDSystemImpl implements PIDSystem {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PurlPIDSystemImpl.class);
+    private static final ValueFactory VALUEFACTORY = SimpleValueFactory.getInstance();
+    private static String fdpUri = "";
+    
+    @Autowired
+    @Qualifier("purlBaseUrl")
+    private String purlBaseUrl;
+
+    /**
+     * Create a new purl.org PID uri for a given metadata
+     *
+     * @param <T>
+     * @param metadata Subtype of Metadata object
+     * @throws NullPointerException exception if the metadata or the metadata URI is null
+     * @throws IllegalStateException exception if the base purl.org is empty
+     * @throws IllegalStateException exception if the fdpUri is empty or null
+     * @return purl.org uri as IRI
+     */
+    @Override
+    public <T extends Metadata> IRI getURI(T metadata) throws IllegalStateException {
+        Preconditions.checkNotNull(metadata, "Metadata must not be null.");
+        Preconditions.checkNotNull(metadata.getUri(), "Metadata URI must not be null.");
+        Preconditions.checkNotNull(purlBaseUrl, "Purl base url can't be null.");
+        Preconditions.checkState(!purlBaseUrl.isEmpty(), "Purl base url can't be empty");
+
+        if (metadata instanceof FDPMetadata) {
+            fdpUri = metadata.getUri().toString();
+        }
+        LOGGER.info("Creating an new purl.org PID");
+        String purlIRI = metadata.getUri().toString().replace(fdpUri, purlBaseUrl);
+        
+        return VALUEFACTORY.createIRI(purlIRI);
+
+    }
+
+    /**
+     * Returns identifier of a given purl PID iri.
+     *
+     * @param iri Purl PID IRI
+     * @throws NullPointerException exception if the purl pid URI is null
+     * @throws IllegalStateException exception if the purl URI doesn't contain "purl.org" string
+     * @return ID as String
+     */
+    @Override
+    public String getId(IRI iri) {
+        Preconditions.checkNotNull(iri, "Purl pid uri must not be null.");
+        Preconditions.checkState(iri.toString().contains("purl.org"),
+                "Not an valid default pid uri.");
+        String id = null;
+        String uri = iri.toString();
+        id = uri.substring(uri.lastIndexOf('/') + 1, uri.length());
+        return id;
```

Comment:
Any reason to choose this (custom) implementation over using [`IRI.getLocalName()`](http://docs.rdf4j.org/javadoc/latest/org/eclipse/rdf4j/model/IRI.html#getLocalName--)?